docs/tools.md

```diff
@@ -90,3 +90,24 @@ El primero es que el estado de mantenimiento de dotenv es mucho mejor que el de
 El segundo es que, mientras que dotenv no tiene dependencias, nconf tiene 4. Cuantas menos dependencias la librería mejor porque menos dependencias tendrá el proyecto.
 El tercero es que dotenv tiene el soporte para tipos totalmente integrado mientras que nconfg las tiene en un paquete a parte(esto es un detalle).
 
+
+## REST API Framework
+
+#### Necesario:
+1. Flexible: Llegado este punto el proyecto ya está estructurado. El framework ha de tener la capacidad de adaptarse a la estructura actual del proyecto.
+2. Eficiente: Ha de ser capaz de gestionar suficientes peticiones por unidad de tiempo.
+3. Integrable: Fundamentalmente con las herramientas escogidas para test y logging
+4. Soporte para tipos de typescript
+
+#### Conveniente:
+1. Métodos Incorporables al Servidor: La capacidad de ligar ciertos métodos al servidor puede ser útil puesto que es una muy buena alternativa a importar un módulo de funciones auxiliares en todos sitios.
+2. Capacidad para Servir Contenido Estático: En ocasiones será necesario servir un simple archivo desde disco, es útil tener la capacidad para hacerlo.
+
+
+Se han considerado los siguientes frameworks: [nest](https://www.npmjs.com/package/@nestjs/core), [fastify](https://www.npmjs.com/package/fastify), [hapi](https://www.npmjs.com/package/@hapi/hapi), [restify](https://www.npmjs.com/package/restify).
+
+- Nest es la alternativa más popular. Nest es un proyecto apoyado por grandes multinacionales como Redhat y empresas como NX y Labster que tiene como objetivo resolver los problemas arquitectónicos frecuentes en las aplicaciones escritas en javascript/typescript: "... However, while plenty of superb libraries, helpers, and tools exist for Node (and server-side JavaScript), none of them effectively solve the main problem of - Architecture". Esto implica que las herramientas necesarias para construir el API están pensadas para usarse con el resto de las herramientas del proyecto pero sobre todo que hay que seguir la estructura de aplicación propuesta. Por lo tanto no verifica la tercera condición y ha sido descartado.
```

Comment:
¿Es flexible, integrable, eficiente y tiene soporte para Typescript? Después de establecer los criterios ni siquiera los mencionas.

Reply:
Ya he respondido a esto arriba...

Reply:
Sí, pero el que te lo pregunte indica que no está nada claro. ¿Qué trabajo os cuesta poner una tabla con vuestros criterios y si se cumplen o no?

Reply:
La frase en inglés está copiada aparentemente de https://www.mitzon.com/mooc/develop-restful-web-services-using-nestjs-and-mongodb/ y ni siquiera tiene sentido. Es una web con mucho *SEO juice* que atrae a los incautos. ¿Qué es la "Arquitectura -"?

*Use the source, Luke*: Si te vas a [ella](https://docs.nestjs.com/v4/), lo que dice es:
> none of them effectively solve the main problem - the architecture.

Lo que sí tiene sentido (y es también irrelevante a su elección o no como framework en este contexto)